pygeoapi/api/stac.py

```diff
@@ -217,6 +234,210 @@ def get_stac_path(api: API, request: APIRequest,
+        'rel': request.get_linkrel(F_JSON),
+        'type': FORMAT_TYPES[F_JSON],
+        'title': l10n.translate('This document as JSON', request.locale),
+        'href': f"{api.base_url}/stac-api?f={F_JSON}"
+    }, {
+        'rel': 'root',
+        'type': FORMAT_TYPES[F_JSON],
+        'title': l10n.translate('This document as JSON', request.locale),
+        'href': f"{api.base_url}/stac-api?f={F_JSON}"
+    }, {
+        'rel': 'service-desc',
+        'type': 'application/vnd.oai.openapi+json;version=3.0',
+        'title': l10n.translate('The OpenAPI definition as JSON', request.locale),  # noqa
+        'href': f"{api.base_url}/openapi"
+    }, {
+        'rel': 'service-doc',
+        'type': FORMAT_TYPES[F_HTML],
+        'title': l10n.translate('The OpenAPI definition as HTML', request.locale),  # noqa
+        'href': f"{api.base_url}/openapi?f={F_HTML}",
+        'hreflang': api.default_locale
+    }, {
+        'rel': 'search',
+        'type': FORMAT_TYPES[F_JSON],
+        'title': l10n.translate('STAC API search', request.locale),
+        'href': f"{api.base_url}/stac-api//search?f={F_JSON}"
+    }]
+
+    return headers, status, to_json(content, api.pretty_print)
+
+
+def search(api: API, request: Union[APIRequest, Any]) -> Tuple[dict, int, str]:
+    """
+    STAC API Queries stac-collection
+
+    :param request: A request object
+    :param dataset: dataset name
+
+    :returns: tuple of headers, status code, content
+    """
+
+    stac_api_collections = {}
+
+    request._format = F_JSON
+
+    headers = request.get_response_headers(**api.api_headers)
+
+    LOGGER.debug('Checking for STAC collections')
+    collections = filter_dict_by_key_value(api.config['resources'],
+                                           'type', 'stac-collection')
+
+    if not collections:
+        return api.get_exception(
+            HTTPStatus.NOT_IMPLEMENTED, headers, F_JSON, 'NotImplemented',
+            'No configured STAC searchable collection')
+
+    LOGGER.debug('Checking for STAC collections with features or records')
+    for key, value in collections.items():
+        found_collection = False
+        for fr in ['feature', 'record']:
+            try:
+                _ = get_provider_by_type(value['providers'], fr)
+                found_collection = True
+                break
+            except ProviderTypeError:
+                pass
+
+        if found_collection:
+            stac_api_collections[key] = value
+
+    if not stac_api_collections:
+        msg = 'No STAC API collections configured'
+        return api.get_exception(HTTPStatus.INTERNAL_SERVER_ERROR, headers,
+                                 request.format, 'NotApplicable', msg)
+
+    if request.data:
+        LOGGER.debug('Intercepting STAC POST request into query args')
+        request_data = json.loads(request.data)
+        request_params = deepcopy(dict(request.params))
+
+        for qp in ['bbox', 'datetime', 'limit', 'offset']:
```

Comment:
I suppose it's a intended limitation to limit query to space/time instead of parse parameters basing on result of queriable entrypoint of the STAC collection/catalog. Am I right? If so IMHO it should be documented and ask for contribution about this aspect.
In any case, great addition :)

Reply:
Thanks for the feedback @luipir.  The PR implements the STAC API Core and Item Search components in STAC API.  I've put forth a PR in #2132 to help clarify for users.

If we need additional functionality for STAC API, this should be put forth in an issue for discussion as a next step.